mautic/mautic
# More rector dead code rules

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [y]
| Deprecations?                          | [n]
| BC breaks? (use the c.x branch)        | [y]
| Automated tests included?              | [/] 
| Related user documentation PR URL      | /
| Related developer documentation PR URL | /
| Issue(s) addressed                     | /



#### Description:

This PR is adding more Rector rules that are removing dead code. These rules will run in our CI so no dead code covered by these rules will be added to the code base in the first place.

Rector pointed out to 2 classes that I couldn't find to be used anywhere so I removed it when we have the chance with M5.0.0.

#### Steps to test this PR:


The code changes are all over the place, but here are some of the affected places that can be tested
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a report
3. Execute a campaign
4. Create a form with the Upload field and test that a file can be uploaded by it
5. View a tag detail page in the tag manager
